### ISSUE TITLE
Like block: handle cases when Likes are disabled

### ISSUE THREAD

**Issue by ivan-ottinger** [Automattic/wp-calypso#85156]
Currently, when the Likes are disabled, we get unexpected behavior of the Like block:

1. If Likes and Reblogs are disabled in Calypso (in _Tools → Marketing → Sharing Buttons_):

![Image](https://github.com/Automattic/wp-calypso/assets/25105483/dcf72417-5c6a-487e-8fd6-62a19255a3d5)

The Like block is stuck at loading as can be seen over here:

![Image](https://github.com/Automattic/wp-calypso/assets/25105483/656f655a-ecde-4473-8050-f0d78d350f74)

2. If Likes are disabled completely as Jetpack module (in _Jetpack → Settings → Sharing_):

![Markup on 2023-12-12 at 16:39:23](https://github.com/Automattic/wp-calypso/assets/25105483/29ccd3ed-988e-4f27-8004-5d7c8e8d1833)

The Like block is stuck at loading as can be seen over here:

![Image](https://github.com/Automattic/wp-calypso/assets/25105483/da424b79-eb81-44e8-bb58-68ace6105de8)

In the cases above, we:

**A) Shouldn't render anything in the front-end**

- can consider displaying information to the user in the block editor - letting them know how to enable the Like block
- can consider hiding the block from the block editor selection of blocks (the block should still stay in the post if it has been added there previously)

-- or --

**B) Make the Like block independent of the above settings completely**

In this case, we will need to make sure the Like block is loading the resources it needs (e.g. `queuehandler.js` and the `master.html` iframe).

Currently, for the Jetpack sites, if the `Likes` module is disabled, `the_content` filter isn't added and hence the `post_like()` never runs. This function is responsible for adding necessary resources for the Like button to load properly.

![Image](https://github.com/Automattic/wp-calypso/assets/25105483/200cf873-e3eb-4224-8d78-ea9b5d1a815d)
(this is all inside `projects/plugins/jetpack/modules/likes.php`)

-- or --

**C) Maybe we could do the following steps instead:**

1. do not decouple the Like block from the Like module - since it is so dependent on it
2. if a post or template includes Like block, hide the Like widget automatically; the reason for this is that there’s currently no way to hide the Like widget without messing with the Like module - unless we introduce some new setting
  - additional issue to consider here is that there’s no easy way to check if post _template_ has certain block in it; we can check only if the post _content_ has the Like block in it currently (using `has_block()` or `parse_blocks()`); Maybe we could use `get_block_templates()` that returns all the block templates and then parse the contents of each - to search for the Like block; but it might be too much to do on every front-page load. 🤔  
3. take the steps listed in the option A as well

**Comment by ivan-ottinger** [comment]
Latest thoughts on the way forward: pdDOJh-2JP-p2#comment-2185.

tl;dr; As a part of this task, let's do the following:

If the Likes module is disabled (WoA and JP sites) or if user disables Likes in _Tools → Marketing → Sharing Buttons_

- make the Like block unavailable 
- don't render the block in the front-end
- if the user already added blocks to their posts consider displaying information in the block editor to let them know how they can re-enable the Like block
